mycroft/util/log.py

```diff
@@ -14,33 +14,116 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with Mycroft Core.  If not, see <http://www.gnu.org/licenses/>.
-import json
+import inspect
 import logging
+import sys
+from traceback import format_exception
 
 from os.path import isfile
+
 from mycroft.util.json_helper import load_commented_json
 
-SYSTEM_CONFIG = '/etc/mycroft/mycroft.conf'
 
-__author__ = 'seanfitz'
+def getLogger(name="MYCROFT"):
+    """Depreciated. Use LOG instead"""
+    return logging.getLogger(name)
 
-log_level = "DEBUG"
 
-if isfile(SYSTEM_CONFIG):
-    config = load_commented_json(SYSTEM_CONFIG)
-    log_level = config.get("log_level", "DEBUG")
+class LOG:
+    """
+    Custom logger class that acts like logging.Logger
+    The logger name is automatically generated by the module of the caller
+
+    Usage:
+        LOG.debug('My message: %s', debug_str)
+        LOG('custom_name').debug('Another message')
+    """
 
-FORMAT = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
-logging.basicConfig(format=FORMAT, level=logging.getLevelName(log_level))
-logger = logging.getLogger("MYCROFT")
+    _custom_name = None
 
+    @classmethod
+    def init(cls):
+        sys_config = '/etc/mycroft/mycroft.conf'
+        level_name = load_commented_json(sys_config)['log_level'] \
+            if isfile(sys_config) else 'DEBUG'
```

Comment:
This is a personal opinion, I find the ternary operation a bit confusing when split on different lines like this.